### ISSUE TITLE
[Popover] Content not discoverable via screen reader browse mode

### ISSUE THREAD

**Issue by svinkle** [Shopify/polaris-react#4419]
## Issue summary

The revealed content for popover components appear visually next to the activator control. However, the content is dynamically injected at the bottom of the DOM, outside of the `main` region landmark.

As a result, when using [virtual/browse navigation mode](https://tink.uk/understanding-screen-reader-interaction-modes/), screen reader users are unable to find the revealed content. For example, when using JAWS or NVDA on Windows, using the ⬇️ key to move to the popover content moves the virtual cursor to the next item on the screen, not the popover content as expected.

In addition, the current implementation includes shifting focus to a `div` container without a label which creates a "group." The user then needs to move "into" the group container to discover content. Some environments miss conveying this detail which results in content being skipped over on ⬇️ key press.

Another addition, <kbd>Esc</kbd> seems to be partially supported. When the popover is dismissed, content is removed from view, yet focus shifts to another area of the screen. This would be unexpected.

<details>
  <summary>Screenshots</summary>

<img width="661" alt="Admin draft order screen. Collect payment button is highlighted with its popover content revealed." src="https://user-images.githubusercontent.com/1392632/130469325-b52de9a1-9592-42cb-8339-38a333cd7ba9.png">

</details>

#### HTML

```html
<main>
  <!-- … -->

  <button
    class="Polaris-Button_r99lw Polaris-Button--primary_7k9zs"
    type="button"
    aria-controls="Polarispopover75"
    aria-owns="Polarispopover75"
    aria-expanded="false"
    tabindex="0"
  >
    <span class="Polaris-Button__Content_xd1mk">
      <span class="Polaris-Button__Text_yj3uv">Collect payment</span>
      <!-- … -->
    </span>
  </button>

  <!-- … -->
</main>

<!-- … -->

<div
  class="
    Polaris-PositionedOverlay_2y1hl
    Polaris-Popover__PopoverOverlay_emiux
    Polaris-Popover__PopoverOverlay--open_4ccww
  "
  style="top: 784px; left: 1065.92px; width: 155.906px"
>
  <!-- … -->
  <div
    id="Polarispopover78"
    tabindex="-1"
    class="Polaris-Popover__Content_xd1mk"
    style="height: 98px"
  >
    <!-- … -->
    <div class="Polaris-ActionList_z4041">
      <div class="Polaris-ActionList__Section--withoutTitle_1vpsz">
        <ul class="Polaris-ActionList__Actions_17m67">
          <li>
            <button type="button" class="Polaris-ActionList__Item_yiyol">
              <span class="Polaris-ActionList__Content_xd1mk">
                <span class="Polaris-ActionList__Text_yj3uv">
                  Pay by credit card
                </span>
              </span>
            </button>
          </li>
          <!-- … -->
        </ul>
      </div>
    </div>
    <!-- … -->
  </div>
  <!-- … -->
</div>
```

## Steps to reproduce

1. Go to the [draft order page](https://shop1.myshopify.io/admin/draft_orders/new) with any supported browser
2. With an active screen reader (JAWS or NVDA for example), use the virtual cursor to navigate to and activate the "Collect payment" activator control
3. Use the ⬇️ key to navigate to the revealed content

### Expected

- ⬇️ key to move the virtual cursor to the "Pay by credit card" control.

### Actual

- Virtual cursor moves to the "Find or create a customer" heading.

## Recommendation

1. Adjust the position of the dynamically injected content to be **below** the activator control in the DOM. With this change, ⬇️ key will move the virtual cursor position to the newly revealed content as expected.
2. On activation **do not shift focus.** The disclosure pattern expects focus to remain on the activator. From this point the user is able to press ⬇️ key and discover content organically. (Remove the `tabindex` attribute from `div#Polarispopover78` container as this is no longer required.)
3. On popover dismiss via <kbd>Esc</kbd> or other means, shift focus back to the activator control. From here the user will be able to navigate forward as expected.

Review the [APG 1.1 Disclosure](https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure) pattern for more details on expected keyboard interaction and screen reader support.

#### HTML

```html
<main>
  <!-- … -->

  <button
    class="Polaris-Button_r99lw Polaris-Button--primary_7k9zs"
    type="button"
    aria-controls="Polarispopover75"
    aria-owns="Polarispopover75"
    aria-expanded="false"
    tabindex="0"
  >
    <span class="Polaris-Button__Content_xd1mk">
      <span class="Polaris-Button__Text_yj3uv">Collect payment</span>
      <!-- … -->
    </span>
  </button>
  <div
    class="
      Polaris-PositionedOverlay_2y1hl
      Polaris-Popover__PopoverOverlay_emiux
      Polaris-Popover__PopoverOverlay--open_4ccww
    "
    style="top: 784px; left: 1065.92px; width: 155.906px"
  >
    <!-- … -->
    <div
      id="Polarispopover78"
      class="Polaris-Popover__Content_xd1mk"
      style="height: 98px"
    >
      <!-- … -->
      <div class="Polaris-ActionList_z4041">
        <div class="Polaris-ActionList__Section--withoutTitle_1vpsz">
          <ul class="Polaris-ActionList__Actions_17m67">
            <li>
              <button type="button" class="Polaris-ActionList__Item_yiyol">
                <span class="Polaris-ActionList__Content_xd1mk">
                  <span class="Polaris-ActionList__Text_yj3uv">
                    Pay by credit card
                  </span>
                </span>
              </button>
            </li>
            <!-- … -->
          </ul>
        </div>
      </div>
      <!-- … -->
    </div>
    <!-- … -->
  </div>

  <!-- … -->
</main>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **OS:** Any
- **WCAG Principal:** Operable
- **WCAG SC:** [2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html) – Level A
- **Severity:** High
- **Effort:** Unknown
